### PR TITLE
Refactor project structure

### PR DESCRIPTION
**Pull request by thiagomuller**
Hey guys, I just refactored project structure to decouple core business code from infra code.
This can be further decoupled, but it's a start;
Please review the code and let me know if you have any suggestions.

### CODE REVIEW

File: src/main/java/io/botcrafting/botcraft/core/service/handler/HelpCommandProcessor.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.botcrafting.botcraft.core.service.handler;
+
+public class HelpCommandProcessor extends CommandMessageProcessor{	
+
+	public HelpCommandProcessor(CommandMessageProcessor nextProcessor) {
+		super(nextProcessor);
+	}
+
+	@Override
+	public boolean processMessage(String message) {
```
Comment (rbsilva) [inline src/main/java/io/botcrafting/botcraft/core/service/handler/HelpCommandProcessor.java:10]: is this a working in progress? this TODO will be implemented later?

**Review by rbsilva — CHANGES_REQUESTED** [review]
Nice job with the refactor, I liked the interfaces for sending messages and to get books. Just some minor fixes that need to be done before merge:

- Remove the JVM error logs and add them to gitignore. 
- Check the TODO.
 
Suggestions:

- Maybe we could get rid of the mappers and have the telegram inbound classes do the conversion by having toSomething methods, like toUser or toChat, that converts to our core classes.

**Comment by thiagomuller** [comment]
Hey Rodrigo,
Thanks for reviewing, just removed the log file and put *.log files on gitignore.
The TODO was from another refactor I'm working on right now, but since it's not yet ready, i took away those files for this PR.
Could you kindly review it again?

Thanks!!

**Review by rbsilva — APPROVED** [review]
looks good

**Pull request merged**